pkg/integrations/cadvisor/cadvisor.go

```diff
@@ -0,0 +1,133 @@
+package cadvisor //nolint:golint
+
+import (
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/google/cadvisor/container"
+
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/integrations/config"
+)
+
+// DefaultConfig holds the default settings for the cadvisor integration
+var DefaultConfig Config = Config{
+	// Common cadvisor config defaults
+	StoreContainerLabels: true,
+	ResctrlInterval:      0,
+	// Matching the default disabled set from cadvisor - https://github.com/google/cadvisor/blob/3c6e3093c5ca65c57368845ddaea2b4ca6bc0da8/cmd/cadvisor.go#L78-L93
+	DisabledMetricsSet: container.MetricSet{
+		container.MemoryNumaMetrics:              struct{}{},
+		container.NetworkTcpUsageMetrics:         struct{}{},
+		container.NetworkUdpUsageMetrics:         struct{}{},
+		container.NetworkAdvancedTcpUsageMetrics: struct{}{},
+		container.ProcessSchedulerMetrics:        struct{}{},
+		container.ProcessMetrics:                 struct{}{},
+		container.HugetlbUsageMetrics:            struct{}{},
+		container.ReferencedMemoryMetrics:        struct{}{},
+		container.CPUTopologyMetrics:             struct{}{},
+		container.ResctrlMetrics:                 struct{}{},
+		container.CPUSetMetrics:                  struct{}{},
+	},
+	EnabledMetricsSet: container.MetricSet{},
+
+	// Containerd config defaults
+	Containerd:          "/run/containerd/containerd.sock",
```

Comment:
Will this break on non-Linux? Do we need to do something similar to windows_exporter to make this Linux-only?

Reply:
It does, indeed on anything non linux. After a couple sloppy attempts with build tags, I *think* got this straight.

Reply:
Your changes for that LGTM 👍